persistence/block.go

```diff
@@ -55,15 +76,51 @@ func (p PostgresContext) TransactionExists(transactionHash string) (bool, error)
 	return true, err
 }
 
-func (p PostgresContext) StoreTransaction(txResult modules.TxResult) error {
-	return p.txIndexer.Index(txResult)
+func (p PostgresContext) StoreTransactions() error {
+	// TODO: store in batch
+	for _, txResult := range p.GetLatestTxResults() {
+		if err := p.txIndexer.Index(txResult); err != nil {
+			return err
+		}
+	}
+	return nil
 }
 
-func (p PostgresContext) StoreBlock(blockProtoBytes []byte) error {
+// DISCUSS: this might be retrieved from the block store - temporarily we will access it directly from the module
+//       following the pattern of the Consensus Module prior to pocket/issue-#315
+func (p *PostgresContext) SetProposalBlock(blockHash string, blockProtoBytes, proposerAddr, qc []byte, transactions [][]byte) error {
+	p.latestBlockHash = blockHash
+	p.latestBlockProtoBytes = blockProtoBytes
```

Comment:
The `blockProtoBytes` are a serialization of the consensus block proto. I don't think we need this. The interface transfers the primitives (address, QC, etc...) from consensus to persistence, and the bytes that'll be stored are those that are a serialization of the persistence specific proto (which I add in the follow up commit).

Reply:
Okay - so what do you suggest? To me, since your comment is based on `future work` it might make sense to amend in that commit. 

```
if err := p.blockstore.Put(heightToBytes(p.Height), p.latestBlockProtoBytes); err != nil {
		return err
	}
``` 

We need to serialize and store for `make compose_and_watch` to work in `this commit`

Reply:
Okay understood. Adding a comment about adding a TODO

Reply:
Leave as is for now. Just need a TODO near the method signature that I'll remove it.